src/libxtp/tools/decomp.cc

```diff
@@ -0,0 +1,65 @@
+/*
+ *            Copyright 2009-2021 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "decomp.h"
+#include <sstream>
+
+// Third party includes
+#include <boost/algorithm/string.hpp>
+
+// Local VOTCA includes
+#include "votca/xtp/activedensitymatrix.h"
+#include "votca/xtp/gaussianwriter.h"
+#include "votca/xtp/orbitals.h"
+#include "votca/xtp/pmdecomposition.h"
+#include <votca/tools/constants.h>
+
+namespace votca {
+namespace xtp {
+
+void Decomp::ParseOptions(const tools::Property& options) {
+  orbitals.ReadFromCpt(_job_name + ".orb");
+  std::string temp = options.get("activeatoms").as<std::string>();
+  std::stringstream ss(temp);
+  Index tmp;
+  while (ss >> tmp) {
+    activeatoms.push_back(tmp);
+  }
+  std::cout << "Atoms in active region: ";
+  for (const auto& atom : activeatoms) {
+    std::cout << atom << " ";
+  }
+  std::cout << std::endl;
```

Comment:
we have a class `IndexParser` for that which does all that for you.

Reply:
https://github.com/votca/xtp/blob/6d8228336c7c730e9fa9dfb93f1c8cef72ed1deb/include/votca/xtp/IndexParser.h